### PR TITLE
Fix .iex.exs error behaviour #8618

### PR DESCRIPTION
**Pull request by f1sty**
Loading .iex.exs with compile or runtime errors currently halts the system. 
This fix intended to print error message, but still run iex with default state.

### CODE REVIEW

**Comment by f1sty** [comment]
Added test for malformed .iex, check if it sufficient, please.

File: lib/iex/test/iex/interaction_test.exs [hunk 1]
```diff
@@ -132,6 +132,15 @@ defmodule IEx.InteractionTest do
     File.rm("dot-iex")
   end
 
+  test "malformed .iex" do
+    File.write!("dot-iex", "malformed")
+    capture_io(:stderr, fn ->
```
[the remark this answers is not in the document]
Comment (f1sty) [reply lib/iex/test/iex/interaction_test.exs]: Seems like it, removed redundancy, thanks!

**Comment by josevalim** [comment]
Thanks @f1sty! Your PR was written on top of a very old master. Can you please rebase master and rebase your PR? After that, make sure to run `make compile && mix format && make test`. Thank you.

**Comment by f1sty** [comment]
> Can you please rebase master and rebase your PR? After that, make sure to run `make compile && mix format && make test`. 

Done;)

**Comment by josevalim** [comment]
@f1sty unfortunately your PR has unrelated changes now, probably because i told you to run `mix format` but you probably should have run `bin/elixir bin/mix format`. Can you please exclude teh unrelated changes from your PR and push?

File: lib/iex/test/iex/interaction_test.exs [hunk 2]
```diff
@@ -180,5 +180,14 @@ defmodule IEx.InteractionTest do
       File.rm("dot-iex-1")
       File.rm("dot-iex")
     end
+
+    test "malformed .iex" do
+      File.write!("dot-iex", "malformed")
+
+      assert capture_iex("malformed", [], dot_iex_path: "dot-iex") =~
+               "** (CompileError) iex:1: undefined function malformed/0"
```
Comment (fertapric) [inline lib/iex/test/iex/interaction_test.exs]: I think this should be `dot-iex` :)

```suggestion
               "** (CompileError) dot-iex:1: undefined function malformed/0"
```


Besides, I would use a different input for the `capture_iex` function so it's clear the error comes from the "dot iex" file.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by f1sty** [comment]
:birthday: Thanks!

**Pull request merged**